language/tools/move-mv-llvm-compiler/src/stackless/rttydesc.rs

```diff
@@ -170,17 +185,45 @@ fn define_type_info_global_nil(
     let ll_ty = llcx.int_type(8);
     let ll_global = llmod.add_global(ll_ty, symbol_name);
     ll_global.set_constant();
+    ll_global.set_linkage(llvm::LLVMLinkage::LLVMPrivateLinkage);
+    ll_global.set_unnamed_addr();
     // just an eye-catching marker value
     let value = 255;
     let ll_const = llvm::Constant::int(ll_ty, U256::from(value as u128));
     ll_global.set_initializer(ll_const);
     ll_global
 }
 
+/// Type info for vectors.
+///
+/// Defined in the runtime by `VectorTypeInfo`.
+fn define_type_info_global_vec(
+    llcx: &llvm::Context,
+    llmod: &llvm::Module,
+    symbol_name: &str,
+    elt_mty: &mty::Type,
+) -> llvm::Global {
+    // A struct containing a pointer to a `MoveType`
+    // type descriptor of the element type.
+    let ll_ty = llcx.get_anonymous_struct_type(&[llcx.int_type(8).ptr_type()]);
+    let ll_global = llmod.add_global(ll_ty, symbol_name);
+    ll_global.set_constant();
+    ll_global.set_linkage(llvm::LLVMLinkage::LLVMPrivateLinkage);
+    ll_global.set_unnamed_addr();
+    let elt_tydesc_ptr = define_llvm_tydesc(llcx, llmod, elt_mty).ptr();
+    let ll_const = llcx.const_struct(&[elt_tydesc_ptr]);
+    ll_global.set_initializer(ll_const);
+    ll_global
+}
+
 fn global_tydesc_name(mty: &mty::Type) -> String {
     use mty::{PrimitiveType, Type};
     let name = match mty {
         Type::Primitive(PrimitiveType::U64) => "u64",
+        Type::Vector(elt_ty) => match **elt_ty {
+            Type::Primitive(PrimitiveType::U64) => "vector_u64",
+            _ => todo!(),
+        },
```

Comment:
Same comments as above for fn `type_name`. We could just to a string replace of the `<` with `_` and `>` with space on strings returned by `type_name`/`Type` formatter.

Reply:
I have made this change except just used underscores for both. This will need to be revisited because it is subject to name collisions, but all our symbol naming will probably need mangling eventually.